packages/gatsby-plugin-catch-links/src/catch-links.js

```diff
@@ -1,81 +1,160 @@
+
+  /**
+   * As per https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target
+   */
+  (
+    anchor.target === `_parent` && (
+      !anchor.ownerDocument.defaultView.parent || 
+      anchor.ownerDocument.defaultView.parent === anchor.ownerDocument.defaultView
+    )
+  ) || 
+  (
+    anchor.target === `_top` && (
+      !anchor.ownerDocument.defaultView.top || 
+      anchor.ownerDocument.defaultView.top === anchor.ownerDocument.defaultView
+    )
+  )
+)
+
+export const authorIsForcingNavigation = anchor => (
+  /**
+   * HTML5 attribute that informs the browser to handle the 
+   * href as a downloadable file; let the browser handle it
+   */
+  anchor.hasAttribute(`download`) === true ||
+
+  /**
+   * Let the browser handle anything that doesn't look like a 
+   * target="_self" anchor
+   */
+  anchorsTargetIsEquivalentToSelf(anchor) === false
+)
+
+// https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy
+export const urlsAreOnSameOrigin = (origin, destination) => (
+  origin.protocol === destination.protocol &&
+
+   /* a.host includes both hostname and port in the expected format host:port */
+  origin.host === destination.host
+)
+
+export const pathIsNotHandledByApp = destination => {
+  const pathStartRegEx = new RegExp(`^${escapeStringRegexp(withPrefix(`/`))}`)
+  const pathFileExtensionRegEx = /^.*\.((?!htm)[a-z0-9]{1,5})$/i
+
+  return (
+    /** 
+     * For when pathPrefix is used in an app and there happens to be a link
+     * pointing to the same domain but outside of the app's pathPrefix. For
+     * example, a Gatsby app lives at https://example.com/myapp/, with the
+     * pathPrefix set to `/myapp`. When adding an absolute link to the same
+     * domain but outside of the /myapp path, for example, <a
+     * href="https://example.com/not-my-app"> the plugin won't catch it and
+     * will navigate to an external link instead of doing a pushState resulting
+     * in `https://example.com/myapp/https://example.com/not-my-app`
+     */
+    pathStartRegEx.test(`${destination.pathname}`) === false ||
+
+    /**
+     * Don't catch links pointed at what look like file extensions (other than
+     * .htm/html extensions).
+     */
+    destination.pathname.search(pathFileExtensionRegEx) !== -1
+  )
+}
+
+export const hashShouldBeFollowed = (origin, destination) => (
+  destination.hash !== `` && (
+    /**
+     * Dynamically created anchor links (href="#my-anchor") do not always 
+     * have pathname on IE 
+     */
+    destination.pathname === `` ||
+
+    /* Don't catch links pointed to the same page but with a hash. */
+    destination.pathname === origin.pathname
+  )
+)
+
+export default function(root, cb) {
+  root.addEventListener(`click`, function(event) {
+    if ( userIsForcingNavigation(event) ) return true
+
+    const clickedAnchor = findClosestAnchor(event.target)
+    if (clickedAnchor == null) return true
+
+    if( authorIsForcingNavigation(clickedAnchor) ) return true
```

Comment:
spacing seems inconsistent here. Should make sure theres a space after the if and no space inside the parens.